assignments/this.js

```diff
@@ -1,26 +1,63 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Window binding allows you to use the 'this' keyword within the block scope if no other binding is attached.
+* 2. Implicit binding is when you put the object on the left of this. It is used only when dot notation is used when invoking a function.
+* 3. New bindings are used with constructor functions to create new properties.
+* 4. Explicit binding of this occurs when .call(), .apply(), or .bind() are used on a function.
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
+global.boyOrGirl = "Are you a boy or girl?";
 
-// code example for Window Binding
+function genderQuestion() {
+  console.log(this.boyOrGirl);
+}
+
+genderQuestion();
 
 // Principle 2
 
 // code example for Implicit Binding
 
+const pokemon = {
+  'name': 'Bulbasaur',
+  'type': 'grass',
+  'offensive': function(atk1, atk2) {
+    console.log( `${this.name} can use ${atk1} and ${atk2}.`);
+  }
+}
+
+pokemon.offensive("Tackle", "Vine Whip");
+
 // Principle 3
 
 // code example for New Binding
 
+function RivalQuestion(rivalName) {
+  this.rivalName = rivalName;
+  this.speak = function() {
+    console.log( `Ah, I remember now that your rival's name is ${this.rivalName}!`);
+  }
+}
+const oak = new RivalQuestion('Butt');
+oak.speak();
+
 // Principle 4
 
-// code example for Explicit Binding
\ No newline at end of file
+// code example for Explicit Binding
+
+const trainer = {
+  'name': 'Brock',
+  'age': 16
+}
+
+const skills = ['cook poke treats', 'admire ladies', 'use frying pans and drying pans.'];
+function introduce(skill1, skill2, skill3) {
```

Comment:
What is a drying pan? Hahah. Good examples of binding!

Reply:
https://www.youtube.com/watch?v=oawUi9s3ENE